src/link/interface.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-constraint */
+/* eslint-disable no-use-before-define */
+import type { MultihashDigest } from '../hashes/interface'
+import type { MultibaseEncoder, MultibaseDecoder, Multibase } from '../bases/interface'
+import type { Phantom, ByteView } from '../block/interface'
+
+
+export type { MultihashDigest, MultibaseEncoder, MultibaseDecoder }
+export type Version = 0 | 1
+
+export type DAG_PB = 0x70
+export type SHA_256 = 0x12
+
+/**
+ * Represents an IPLD link to a specific data of type `T`.
+ *
+ * @template T - Logical type of the data being linked to.
+ * @template C - multicodec code corresponding to a codec linked data is encoded with
+ * @template A - multicodec code corresponding to the hashing algorithm of the CID
+ * @template V - CID version
+ */
+export interface Link<
+  Data extends unknown = unknown,
+  Format extends number = number,
+  Alg extends number = number,
+  V extends Version = 1
+  > extends Phantom<Data> {
+  readonly version: V
+  readonly code: Format
+  readonly multihash: MultihashDigest<Alg>
+
+  readonly byteOffset: number
+  readonly byteLength: number
+  readonly bytes: ByteView<Link<Data, Format, Alg, V>>
+
+
+  equals(other: unknown): other is Link<Data, Format, Alg, Version>
+
+  toString<Prefix extends string>(base?: MultibaseEncoder<Prefix>): ToString<Link<Data, Format, Alg, Version>, Prefix>
+  toJSON(): { version: V, code: Format, hash: Uint8Array }
```

Comment:
do we really need this here? maybe this could just be reserved for the concrete `CID` to leave the interface a little simpler.

Reply:
We could omit. Although I think it’s useful and is used in few places where you could accept Link type